lib/Event/ReorderEvent.php

```diff
@@ -30,7 +32,7 @@ class ReorderEvent extends AbstractDocumentEvent
      */
     public function __construct($document, $destId, $after)
     {
-        parent::__construct($document);
+        $this->document = $document;
```

Comment:
No, it was not something that was "designed", it just evolved like this
and, no, it isn't very satisfying.  But I don't think it is important to
change this. Introducing traits and interfaces would be over-the-top in
this situation. Events are not polymorphic (i.e.  interfaces not
required). Traits might be useful but I don't think its worth the
effort.

On Mon, Jul 20, 2015 at 02:06:05PM -0700, Daniel Rotter wrote:
>    In [1]lib/Event/ReorderEvent.php:
> 
>  > @@ -30,7 +32,7 @@ class ReorderEvent extends AbstractDocumentEvent
>  >       */
>  >      public function __construct($document, $destId, $after)
>  >      {
>  > -        parent::__construct($document);
>  > +        $this->document = $document;
> 
>    Because I also changed the class it extends from, and the new one doesn't
>    have a constructor. Don't like this inheritance structure at all, would
>    prefer to replace that with interfaces and traits, but this seemed to be
>    the cleanest solution for now.
> 
>    —
>    Reply to this email directly or [2]view it on GitHub.
> 
>    Reverse link: [3]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/sulu-io/sulu-document-manager/pull/28#discussion_r35044050
>    2. https://github.com/sulu-io/sulu-document-manager/pull/28/files#r35044050
>    3. https://github.com/sulu-io/sulu-document-manager/pull/28/files#r35044050